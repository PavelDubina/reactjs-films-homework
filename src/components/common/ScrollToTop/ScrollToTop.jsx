/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import React, { useEffect, useState } from 'react'
import { faArrowAltCircleUp } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import styles from './ScrollToTop.scss'

export function ScrollToTop() {
  const [isVisible, setIsVisible] = useState(false)

  const toggleVisibility = () => {
    if (window.scrollY > 300) {
      setIsVisible(true)
    } else {
      setIsVisible(false)
    }
  }

  const scrollToTop = () => window.scrollTo({ top: 0, behavior: 'smooth' })

  useEffect(() => {
    window.addEventListener('scroll', toggleVisibility)
  }, [])

  return (
    <div className={styles.scroll_to_top}>
      {isVisible && <FontAwesomeIcon icon={faArrowAltCircleUp} onClick={scrollToTop} />}
    </div>
  )
}
