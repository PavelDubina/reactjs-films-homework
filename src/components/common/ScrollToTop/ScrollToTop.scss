/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

.scroll_to_top {
  font-size: 48px;
  position: fixed;
  right: 0;
  bottom: 10px;
  color: rgba(24, 183, 236, 0.7);
  cursor: pointer;
  z-index: 30;
}
