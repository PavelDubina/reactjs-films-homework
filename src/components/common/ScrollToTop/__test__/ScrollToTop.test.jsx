/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import React from 'react'
import { create } from 'react-test-renderer'
import { ScrollToTop } from '../ScrollToTop'

describe('ScrollToTop', () => {
  let component

  beforeEach(() => {
    global.scrollTo = jest.fn()
    component = create(<ScrollToTop />)
  })

  it('Should render and match snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
