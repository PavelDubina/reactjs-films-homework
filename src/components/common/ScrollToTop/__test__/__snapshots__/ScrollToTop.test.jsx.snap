// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop Should render and match snapshot 1`] = `
<div
  className="scroll_to_top"
/>
`;
